core/infra/config/peer_service_with_json.cpp

```diff
@@ -47,18 +62,22 @@ namespace config {
 
     std::vector<std::unique_ptr<peer::Node>> PeerServiceConfig::getPeerList() {
         std::vector<std::unique_ptr<peer::Node>> nodes;
-        if (auto config = openConfig(getConfigName())) {
-            for (const auto& peer : (*config)["group"].get<std::vector<json>>()){
-                nodes.push_back(std::make_unique<peer::Node>(
-                    peer["ip"].get<std::string>(),
-                    peer["publicKey"].get<std::string>(),
-                    1
-                ));
-            }
+        for( auto node : peerList ) {
+            nodes.push_back( std::make_unique<peer::Node>( node.getIP(), node.getPublicKey(), node.getTrustScore() ) );
         }
+        sort( nodes.begin(), nodes.end(),
+            []( const std::unique_ptr<peer::Node> &a, const std::unique_ptr<peer::Node> &b ) { return a->getTrustScore() > b->getTrustScore(); } );
         return nodes;
     }
 
+    void PeerServiceConfig::addPeer( peer::Node peer ) {
+        peerList.insert( peer );
+    }
+
+    void PeerServiceConfig::removePeer( peer::Node peer ) {
```

Comment:
`peer::Node` will be copied here. 

As it should be?

r-value could be used:
```
peer::Node &&peer
```